### PR TITLE
add export csv button for short circuit analysis results

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 1]
```diff
@@ -358,18 +381,72 @@ const ShortCircuitAnalysisResultTable: FunctionComponent<
...
-                overlayNoRowsTemplate={message}
-            />
+        <Box sx={styles.gridContainer}>
+            <Box sx={styles.csvExport}>
+                <Box style={{ flexGrow: 1 }}></Box>
+                <Box>
+                    <FormattedMessage id="MuiVirtualizedTable/exportCSV" />
+                </Box>
+                <Box>
+                    <IconButton
```
Comment (thangqp) [inline src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx:430]: A new button component arrived after merging export csv Security Analysis, see
https://github.com/gridsuite/gridstudy-app/blob/main/src/components/results/securityanalysis/security-analysis-table.tsx#L87
In the PR of Seddik, we begin using this shared button component, see 

https://github.com/gridsuite/gridstudy-app/pull/1835/files#diff-7d20d0abfbb177dd58fc68d9ebca55723faeac7d98c2c41dbd67adbc1d4cd4faR175

Comment (ghazwarhili) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx]: refacto is done

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 2]
```diff
@@ -355,12 +363,16 @@ const ShortCircuitAnalysisResultTable: FunctionComponent<
         shortCircuitAnalysisStatus,
         !isFetching
     );
-    const rowsToShow = getRows(rows, shortCircuitAnalysisStatus);
+    const rowsToShow = (rows: any) => {
+        let currentRows = getRows(rows, shortCircuitAnalysisStatus);
+        onRowDataUpdated && onRowDataUpdated(currentRows.length === 0);
+        return currentRows;
+    };
```
Comment (thangqp) [inline src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx:370]: Do not need modify code to invoke onRowDataUpdated, it is a callback method of aggrid

File: src/translations/fr.json [hunk 3]
```diff
@@ -487,6 +487,7 @@
     "descTransient": "Transitoire",
     "securityAnalysisError": "Une erreur est survenue lors de l'exécution de l'analyse de sécurité",
     "securityAnalysisCsvResultsError": "Une erreur est survenue lors de l'export des résultats de l'analyse de sécurité",
+    "shortCircuitAnalysisCsvResultsError": "Une erreur est survenue lors de l'export des résultats de court-circuit",
```
Comment (thangqp) [inline src/translations/fr.json:490]: l'analyse de

**Review by thangqp — CHANGES_REQUESTED** [review]

File: src/translations/fr.json (hunk 3 above)
Comment (ghazwarhili) [reply src/translations/fr.json]: done

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx (hunk 2 above)
Comment (ghazwarhili) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx]: done

**Review by thangqp — APPROVED** [review]
Code OK
Test OK

File: src/components/results/shortcircuit/shortcircuit-analysis-result-tab.tsx [hunk 4]
```diff
@@ -108,6 +121,17 @@ export const ShortCircuitAnalysisResultTab: FunctionComponent<
         delay: RESULTS_LOADING_DELAY,
     });
 
+    const handleGridColumnsChanged = useCallback((params: any) => {
+        if (params?.api) {
+            debugger;
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-tab.tsx]: yes

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 5]
```diff
@@ -44,6 +44,8 @@ interface ShortCircuitAnalysisResultProps {
     filterProps: FilterPropsType;
     sortProps: SortPropsType;
     filterEnums: FilterEnumsType;
+    onGridColumnsChanged: (params: any) => void;
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx]: done

**Pull request merged**